tools-internal/kdc-api-server/server/server.go

```diff
@@ -27,18 +31,61 @@ type KDCRequestListPrincipals struct {
 	Binary *bool  `json:"binary"`
 }
 
+type KDCListPrincipalsResponse struct {
+	List     []KPrincipal `json:"list,omitempty"`
+	Checksum string       `json:"checksum,omitempty"`
+}
+
 type KDCCheckStatus struct {
-	Pass   bool   `json:"pass"`
-	Reason string `json:"reason,omitempty"`
+	Pass     bool   `json:"pass"`
+	Reason   string `json:"reason,omitempty"`
+	Checksum string `json:"checksum,omitempty"`
 }
 
 type KDCResponse struct {
-	Status     string          `json:"status"`
-	Error      string          `json:"error,omitempty"`
-	Principals []KPrincipal    `json:"principals,omitempty"`
-	Check      *KDCCheckStatus `json:"check,omitempty"`
+	Status     string                     `json:"status"`
+	Error      string                     `json:"error,omitempty"`
+	Principals *KDCListPrincipalsResponse `json:"principals,omitempty"`
+	Check      *KDCCheckStatus            `json:"check,omitempty"`
+}
+
+/**
+ * Sortable list of principals
+ */
+type KPrincipalList []KPrincipal
+
+func (p KPrincipalList) Len() int {
+	return len(p)
+}
+func (p KPrincipalList) Less(i, j int) bool {
+	return p[i].Full() < p[j].Full()
+}
+func (p KPrincipalList) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+/**
+ * Sortable list of encryption keys
+ */
+type KTKey struct {
+	Type         int32
+	HashContents string
 }
+type KTKeyList []KTKey
 
+func (p KTKeyList) Len() int {
+	return len(p)
+}
+func (p KTKeyList) Less(i, j int) bool {
+	return p[i].Type < p[j].Type
+}
+func (p KTKeyList) Swap(i, j int) {
+	p[i], p[j] = p[j], p[i]
+}
+
+/**
+ * createKDCAPIServer creates a configured instance of a server interface
```

Comment:
I'd remove this comment, this is a non-exported function, and the name explains the purpose well enough...